frontend/docsettings.lua

```diff
@@ -75,6 +76,16 @@ function DocSettings:getSidecarDir(doc_path, force_location)
     local location = force_location or G_reader_settings:readSetting("document_metadata_folder", "doc")
     if location == "dir" then
         path = DOCSETTINGS_DIR..path
+    elseif location == "hash" then
+        -- local doc = document:extend()
+        local file = io.open(doc_path, 'rb')
+        if not file then return '' end
+        local hsh = util.partialMD5(file)
+        -- local hsh = doc:partialMD5(file)
+        file:close()
+        -- converts b3fb8f4f8448160365087d6ca05c7fa2 to b3/fb/ to avoid too many files in one dir
+        local subpath = string.format("/%s/%s/", hsh:sub(1, 2), hsh:sub(3, 4)) --, hsh:sub(5, 6)) -- 3=overkill?
```

Comment:
I think 2 is already overkill :)
1 level would be enough, 256 subdirs will contain at most each a few dozen of sdr dirs. And even 256 * 1000 = 256000 opened books should be fine.

Reply:
I did too, but the example gave 3 so I compromised lol. I'll make it a single directory deep.